src/web/components/ApiKeyManagement/KeyHelper.ts

```diff
@@ -0,0 +1,13 @@
+import { ApiKeyDTO } from '../../../api/services/adminServiceHelpers';
+
+const yearInSeconds = 31556952;
+
+export const needsRotating = (apiKey: ApiKeyDTO): boolean => {
+  const currentDate = new Date().getTime();
+  // apiKey date is in seconds, current date is in milliseconds
+  const currentDateFormat = Math.floor(currentDate / 1000);
+  if (currentDateFormat - apiKey.created > yearInSeconds) {
+    return true;
+  }
+  return false;
+};
```

Comment:
Went to add a test file but unsure how to test a function like this that uses current date that is always changing

Reply:
Should be possible - I think we can just build in enough tolerance to the test. e.g. for the case where the key is older than a year, just make the key 366 days old:
```
/* eslint-disable camelcase */
import { ApiRoleDTO } from '../../../api/entities/ApiRole';
import { ApiKeyDTO } from '../../../api/services/adminServiceHelpers';
import { shouldRotateApiKey } from './KeyHelper';

const Mapper: ApiRoleDTO = { id: 1, roleName: 'MAPPER', externalName: 'Mapper', order: 1 };
const Generator: ApiRoleDTO = {
  id: 2,
  roleName: 'GENERATOR',
  externalName: 'Generator',
  order: 2,
};
const Bidder: ApiRoleDTO = {
  id: 3,
  roleName: 'ID_READER',
  externalName: 'Bidder',
  order: 4,
};
const Sharer: ApiRoleDTO = { id: 4, roleName: 'SHARER', externalName: 'Sharer', order: 3 };

const buildApiKey = (apiKey: Partial<ApiKeyDTO>): ApiKeyDTO => {
  return {
    contact: apiKey.contact ?? 'ApiKey',
    name: apiKey.name ?? 'ApiKey',
    created: apiKey.created ?? 1702830516,
    key_id: apiKey.key_id ?? 'F4lfa.fdas',
    site_id: apiKey.site_id ?? 1,
    disabled: apiKey.disabled ?? false,
    roles: apiKey.roles ?? [Mapper, Generator, Bidder, Sharer],
    service_id: apiKey.service_id ?? 0,
  };
};

describe('does api key need rotating', () => {
  const yearInSeconds = 31556952;
  const dayInSeconds = 60 * 60 * 24;

  it('should return true if the API key is older than one year', () => {
    const apiKey = { created: Math.floor(Date.now() / 1000) - yearInSeconds - dayInSeconds };
    expect(shouldRotateApiKey(buildApiKey(apiKey))).toBe(true);
  });

  it('should handle the case where the API key was just created', () => {
    const apiKey = { created: Math.floor(Date.now() / 1000) };
    expect(shouldRotateApiKey(buildApiKey(apiKey))).toBe(false);
  });
});
```

Note there's some code here that should be extracted out to a new file and used elsewhere in the codebase, but that should be a good start anyway.

Reply:
Thanks - this was super helpful.  I've exported a few things and added two more test cases